libraries/chain/host_context.cpp

```diff
@@ -98,20 +98,23 @@ int64_t host_context::execute_sync_call(name call_receiver, uint64_t flags, std:
 
          const account_metadata_object* receiver_account = &db.get<account_metadata_object, by_name>( call_receiver);
          if (receiver_account->code_hash.empty()) {
-            return handle_call_failure();
+            return handle_call_failure(static_cast<int64_t>(call_error_code::empty_receiver));
          }
 
          const code_object* const codeobject = db.find<code_object, by_code_hash>(boost::make_tuple(receiver_account->code_hash, receiver_account->vm_type, receiver_account->vm_version));
          if (!codeobject || !codeobject->sync_call_supported) {
-            return handle_call_failure();
+            return handle_call_failure(static_cast<int64_t>(call_error_code::sync_call_not_supported_by_receiver));
          }
 
          // use a new sync_call_context for next sync call
          sync_call_context call_ctx(control, trx_context, ordinal, get_current_action_trace(), get_sync_call_sender(), call_receiver, receiver_account->is_privileged(), depth, is_next_call_read_only, data);
 
          try {
             // execute the sync call
-            control.get_wasm_interface().execute(receiver_account->code_hash, receiver_account->vm_type, receiver_account->vm_version, call_ctx);
+            auto status = control.get_wasm_interface().execute(receiver_account->code_hash, receiver_account->vm_type, receiver_account->vm_version, call_ctx);
+            if (status < 0) {
+               return handle_call_failure(status);
+            }
```

Comment:
There isn't any validation that only `-2` or `-3` are returned. It seems given the current design it really ought to be enforced.